server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/management/DelayedSingleRetryRequestExecutorTest.java

```diff
@@ -0,0 +1,117 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.management;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.nurkiewicz.asyncretry.AsyncRetryExecutor;
+
+class DelayedSingleRetryRequestExecutorTest {
+
+    private DelayedSingleRetryRequestExecutor testee;
+
+    @BeforeEach
+    void setUp() {
+        testee = new DelayedSingleRetryRequestExecutor(new AsyncRetryExecutor(Executors.newSingleThreadScheduledExecutor()));
+    }
+
+    @Test
+    void executeShouldReturnSucceededValue() {
+        Integer expected = 3;
+        Integer actual = testee.execute(() -> expected);
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void executeShouldReturnSucceededValueAfterOneFailure() {
+        AtomicInteger atomicInteger = new AtomicInteger();
+        Integer expected = 3;
+        Integer actual = testee.execute(() -> {
+            if (atomicInteger.incrementAndGet() == 2) {
+                return expected;
+            }
+            throw new RuntimeException();
+        });
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void executeShouldThrowAfterTwoFailure() {
```

Comment:
s/Failure/Failures/

Reply:
I don't see the two failures in this test, but maybe there is no simple way to show them